Core/COEditingContext.m

```diff
@@ -412,8 +416,23 @@ - (void)updateCrossPersistentRootReferencesToPersistentRoot: (COPersistentRoot *
 {
 	NSParameterAssert(aPersistentRoot != nil);
 	// NOTE: -delete/undeleteBranch: enforce !aBranch.isCurrentBranch already.
-	NSParameterAssert(aBranch == nil
-		|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	// NOTE: This is used by lazy loading now, and the following assertion no longer holds.
+	// Is this a problem?
+	//NSParameterAssert(aBranch == nil
+	//	|| (!aBranch.isCurrentBranch && aBranch.persistentRoot == aPersistentRoot));
+	
+	// See documentation above
+	if (isDeletion)
+	{
+		// TODO: For an uncommitted persistent root and branch, could be better if
```

Comment:
For this TODO, we can rewrite -deletePersistentRoot like that:

```
- (void)deletePersistentRoot: (COPersistentRoot *)aPersistentRoot
{
    if ([_persistentRootsPendingUndeletion containsObject: aPersistentRoot])
    {
        [_persistentRootsPendingUndeletion removeObject: aPersistentRoot];
    }
    else
    {
        // NOTE: Deleted persistent roots are removed from the cache on commit.
        [_persistentRootsPendingDeletion addObject: aPersistentRoot];
    }

	[self updateCrossPersistentRootReferencesToPersistentRoot: aPersistentRoot
	                                                   branch: nil
	                                                isDeleted: YES];
	
	if ([aPersistentRoot isPersistentRootUncommitted])
    {
		[_persistentRootsPendingDeletion removeObject: aPersistentRoot];
        [self unloadPersistentRoot: aPersistentRoot];
    }
}
```

All the tests still pass when I apply this change. We could do a similar change to -deleteBranch:. What do you think?